### PR TITLE
Change for renderer API integration

### PR DESCRIPTION
**Pull request by pbalint**
This change adds a save button, controllable by a query parameter and glue code so zap is usable by studio v6

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -34,16 +34,33 @@ limitations under the License.
...
 const storage = require('./util/storage.js')
 
+window.addEventListener('message', (event) => {
+    const eventData = event?.data?.eventData
+    switch (event?.data?.eventId) {
+      case 'theme':
+        window[rendApi.GLOBAL_SYMBOL_EXECUTE](rendApi.id.setDarkTheme, eventData.theme === 'dark')
+        break
+      case 'openFile':
+        window[rendApi.GLOBAL_SYMBOL_EXECUTE](rendApi.id.open, eventData.path)
```
Comment (paulr34) [inline src/App.vue:52]: we don't open files with this method anymore

**Review by paulr34 — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 2]
```diff
@@ -206,6 +223,13 @@ export default defineComponent({
           this.$store.dispatch('zap/updateSelectedUcComponentState', resp)
         }
       )
+
+      this.$onWebSocket(
+        dbEnum.wsCategory.dirtyFlag,
```
Comment (paulr34) [inline src/App.vue:228]: The dirty flag is constantly polled from the backend already via web socket

File: src/components/ZCLToolbar.vue [hunk 3]
```diff
@@ -250,6 +264,11 @@ export default {
         )
       }
     },
+    saveChanges() {
```
Comment (paulr34) [inline src/components/ZCLToolbar.vue:267]: this is already done in menu.js

File: src/store/zap/actions.js [hunk 4]
```diff
@@ -817,6 +817,10 @@ export function updateSelectedUcComponentState(context, projectInfo) {
   })
 }
 
+export function setDirtyState(context, isDirty) {
```
Comment (paulr34) [inline src/store/zap/actions.js:820]: again, the polling mechanism already handles the dirty state

File: src/components/ZCLToolbar.vue (hunk 3 above)
Comment (pbalint) [reply src/components/ZCLToolbar.vue:267]: These changes are done in order to embed ZAP into Studio v6, where there's no electron menu...

File: src/App.vue (hunk 2 above)
Comment (pbalint) [reply src/App.vue:228]: Yes, but there's no reflection of that in the UI state, so we can't conditionally show HTML UI elements based on that.
(These changes are done in order to embed ZAP into Studio v6, where there's no electron menu and potentially multiple ZAP UI instances in the same browser instance (sharing the session id)...)

File: .husky/.gitignore [hunk 5]
```diff
@@ -1 +0,0 @@
-_
```
Comment (brdandu) [inline .husky/.gitignore:1]: Get rid of this change

File: .husky/pre-commit [hunk 6]
```diff
@@ -1,17 +0,0 @@
...
-npx jsdoc src-electron src-shared -r -d ./generated-html/
-
-echo "🚀   > Eslint..."
-npx eslint --ext .js,.vue src src-electron src-shared src-script test
-
-echo "🚀   > Reset version in package.json to 0.0.0..."
-src-script/zap-update-package-version.js -fake
-
-echo "🚀   > Check versions in package-lock.json..."
-src-script/zap-validate-metafiles.js
```
Comment (brdandu) [inline .husky/pre-commit:17]: Why is this removed?

File: package.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "zap",
-  "version": "0.0.0",
+  "version": "2023.11.21",
```
Comment (brdandu) [inline package.json:3]: This should not change

File: src/boot/ws.js [hunk 8]
```diff
@@ -24,6 +24,7 @@ import * as Util from '../util/util.js'
 const tickInterval = 15000 // 15 seconds tick interval for server watchdog.
 
 let eventEmitter = new Events.EventEmitter()
+
```
Comment (brdandu) [inline src/boot/ws.js:27]: Get rid of this

File: .husky/pre-commit (hunk 6 above)
Comment (pbalint) [reply .husky/pre-commit]: Wanted to quickly push draft PR, reverted this once the draft was finalized

**Review by paulr34 — APPROVED** [review]
